pulsar-common/src/main/proto/PulsarApi.proto

```diff
@@ -230,6 +230,9 @@ message CommandProducer {
 	/// If a producer name is specified, the name will be used,
 	/// otherwise the broker will generate a unique name
 	optional string producer_name = 4;
+
+	/// Add optional metadata key=value to this producer
+	repeated KeyValue metadata    = 5;
```

Comment:
@merlimat do i need to add optional to this definition?

Reply:
I think `optional` should be added for compatibility of other version.

Reply:
`repeated` is an optional field in protobuf. it means this field can be repeated any number of times (including zero). I don't think you can use `optional` `repeated` together. in proto2 syntax, a field has to be one of 'required', 'optional' and 'repeated'.